client/components/editor-standalone/authoring-storage-in-memory.ts

```diff
@@ -1,17 +1,31 @@
 import ng from 'superdesk-core/scripts/core/services/ng';
 import {IAuthoringAutoSave, IAuthoringStorage} from 'superdesk-api';
 import {getProfile} from './profile';
+import {autosave} from '../../api/autosave';
 
-export function getAuthoringStorageInMemory<T>(
+export function getAuthoringStorageInMemory<T extends {_id: string}>(
     profile: 'event' | 'planning',
     item: T,
     onSave: (current: T, original: T) => Promise<T>,
 ): IAuthoringStorage<T> {
+    /**
+     * Timeout for 500 seconds to let newly added embedded events finish autosaving.
```

Comment:
Timing can't be relied upon. We should do a better solution. Could you elaborate what issue you're trying to address here?

Why did you drop my usage of spread operator to handle validation?